### PR TITLE
golangci-lint: enable goimports and gofumpt

### PR DESCRIPTION
**Pull request by mmorel-35**
**- What I did**
Configure local-prefixes for goimports and add gofumpt linter to golangci-lint
**- How I did it**
Apply  :
* goimports -w -local github.com/docker
* gofumpt -l -w .

**- How to verify it**
Tests are valid 
**- Description for the changelog**
golangci-lint: enable goimports and gofumpt

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: .golangci.yml
Comment (thaJeztah) [inline .golangci.yml:40]: We should probably make this `github.com/docker/docker` to have _actual_ local package only (also see my other comment)

File: .golangci.yml
Comment (thaJeztah) [inline .golangci.yml:131]: There was a bit of a back-and-forth on this one. I'm personally a big fan of `gofumpt`, but there were some concerns about enforcing it, because it may not be what contributors have installed, and enabling it as default formatter may be a bit of a PITA if you contribute to multiple projects (some may be using `gofumpt`, others may not, which could cause an over-enthusiastic IDE to reformat unrelated files). If we _do_ start enforcing it, perhaps we need a `make format` (e.g.) target that can run `gofumpt` containerised (using a `Dockerfile` and `docker build --output` or similar to write the results).

- https://github.com/moby/moby/pull/43169#issuecomment-1018324487

(Perhaps opinions have changed though)

File: builder/builder-next/controller.go
Comment (thaJeztah) [inline builder/builder-next/controller.go:59]: Hm.. it looks like formatting and linting is not (yet) working as expected. ISTR having similar issues in the past when I tried a similar change; in this file, it's creating _more_ groups than the expected 3 (stdlib, external, local);


```go
import (
	"context"
// ....
	"time"

	ctd "github.com/containerd/containerd"
	"github.com/containerd/containerd/content/local"
// ....
	"go.opentelemetry.io/otel/sdk/trace"

	"github.com/docker/docker/api/types"
	"github.com/docker/docker/api/types/filters"
// ....
	"github.com/docker/docker/daemon/graphdriver"
	units "github.com/docker/go-units"

	"github.com/moby/buildkit/solver/pb"
	"github.com/moby/buildkit/util/apicaps"
)
```


I see it also grouped `github.com/docker/go-units` together with `github.com/docker/docker/....` imports (left a comment about that above).

I think the same applies to this changes as `gofumpt`; if we start to enforce this, ideally, we'd also have a `make` target (which could be together with the `gofumpt` one) that performs formatting with the correct options set, so that contributors can verify that all files are formatted correctly, and use that target to correct issues (if there are any), without having to modify their tooling / configuration.

**Review by thaJeztah — CHANGES_REQUESTED** [review]
Thanks for contributing (and apologies for going silent on this one). I generally like the change to be more consistent on formatting; it looks like there's still some things to look into though.

As this is touching _many_ files, changes like this can be somewhat disruptive (likely introducing merge-conflicts in other PRs), so we'd have to decide if we want to go ahead with this one, and pick a right moment to rip-off the band-aid.

I'd like to discuss this with a wider group of the maintainers to see if there's objections. And before we merge, we should probably look at the "tooling" part of this (having (containerized) `make` targets to allow contributors to apply the correct formatting).

File: builder/builder-next/controller.go
Comment (mmorel-35) [reply builder/builder-next/controller.go:59]: Hi @thaJeztah ,
I have tried gci and it seems to be working quite better. See [this PR ](https://github.com/moby/moby/pull/46880) .

**Comment by mmorel-35** [comment]
Superseded by #46880

**Pull request closed without merging**
